src/NumericalAlgorithms/Spectral/BasisFunctions/Jacobi.cpp

```diff
@@ -0,0 +1,53 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "NumericalAlgorithms/Spectral/BasisFunctions/Jacobi.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+namespace Spectral {
+template <typename T>
+T Jacobi::basis_function_value(const double alpha, const double beta,
+                               const size_t k, const T& x) {
+  ASSERT(alpha > -1.0, "alpha = " << alpha);
+  ASSERT(beta > -1.0, "beta = " << beta);
+  // P_0 = 1
+  T P_k_minus_2 = make_with_value<T>(x, 1.);
+  if (k == 0) {
+    return P_k_minus_2;
+  }
+  // P_k_1 = (2+a+b)/2*x + (a-b)/2
+  T P_k_minus_1 = 0.5 * (2.0 + alpha + beta) * x + 0.5 * (alpha - beta);
+  if (k == 1) {
+    return P_k_minus_1;
+  }
+  T P_k = make_with_value<T>(x, 0.);
+  // 2(n+1)(n+a+b+1)(2n+a+b)P_{n+1} - [(2n+a+b+1)(a-b)(a+b) +
+  //  (2n+a+b)(2n+a+b+1)(2n+a+b+2)x]P_n + 2(n+a)(n+b)(2n+a+b+2)P_{n-1} = 0
```

Comment:
Change `n+1`->`n`, etc. and then add:
We solve the recurrence relation by defining the coefficients as `c_0 P_n - (c_1 + c_2 x) P_{n-1} - c_3 P_{n-2}`.

since this is what you actually implemented.